ui/lib/helper.ts

```diff
@@ -201,6 +205,69 @@ export const downloadComplianceCsv = async (
   });
 };
 
+export const downloadThreatscoreReport = async (
+  scanId: string,
+  toast: ReturnType<typeof useToast>["toast"],
+): Promise<void> => {
+  const result = await getThreatscoreReport(scanId);
+
+  if (result?.pending) {
+    toast({
+      title: "The report is still being generated",
+      description: "Please try again in a few minutes.",
+    });
+    return;
+  }
+
+  if (result?.success && result.data) {
+    try {
+      const binaryString = window.atob(result.data);
```

Comment:
This logic is already included in the `downloadComplianceCsv` function.  
Maybe we could extract the common logic into a helper function and pass the type as a prop,  since your type here is `application/pdf` and the type for compliances is `text/csv`.

What do you think?